src/android/Mapbox.java

```diff
@@ -168,7 +183,23 @@ public void run() {
             FrameLayout.LayoutParams params = new FrameLayout.LayoutParams(webViewWidth - left - right, webViewHeight - top - bottom);
             params.setMargins(left, top, right, bottom);
             mapView.setLayoutParams(params);
-
+			//@anothar we have to override back button or app will just close
+            mapView.setOnKeyListener(new View.OnKeyListener() {
```

Comment:
> And pressing the back button on Android should actually close the app. 
No and yes. You are right that your app should decide behaviour. The problem is that you can't define behaviour from your js code without this fix! Without this fix your app always closes and you can't override it. You can't show previous screen to user and you can't override it from js withous this fix. With this fix you can set behaviour to which you want from your js code. This was the real problem for me. Because my app contains many screens and map is not the first one.